### PR TITLE
Always show FFmpeg path in Transcoding page

### PR DESCRIPTION
**Pull request by ploughpuff**
1) Removed the 'hide' and 'required' settings from the html to always show the FFmpeg path.

2) Allow user to submit empty FFmpeg path from Transcoding path to facilitate removal of Custom path without requiring a system restart.

Associated changes on Jellyfin:
https://github.com/jellyfin/jellyfin/pull/844

### CODE REVIEW

**Comment by JustAMan** [comment]
IMHO this should wait for https://github.com/jellyfin/jellyfin/pull/844

**Comment by ploughpuff** [comment]
@JustAMan This PR should be merge too, now that jellyfin/jellyfin#844 has gone.

File: src/scripts/encodingsettings.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ define(["jQuery", "loading", "globalize", "dom"], function($, loading, globalize
...
+        }), page.querySelector("#chkHardwareEncoding").checked = config.EnableHardwareEncoding, $("#selectVideoDecoder", page).val(config.HardwareAccelerationType), $("#selectThreadCount", page).val(config.EncodingThreadCount), $("#txtDownMixAudioBoost", page).val(config.DownMixAudioBoost), page.querySelector(".txtEncoderPath").value = config.EncoderAppPathDisplay || "", $("#txtTranscodingTempPath", page).val(config.TranscodingTempPath || ""), $("#txtVaapiDevice", page).val(config.VaapiDevice || ""), page.querySelector("#selectH264Preset").value = config.H264Preset || "", page.querySelector("#txtH264Crf").value = config.H264Crf || "", page.querySelector("#chkEnableSubtitleExtraction").checked = config.EnableSubtitleExtraction || !1, page.querySelector("#selectVideoDecoder").dispatchEvent(new CustomEvent("change", {
```
Comment (JustAMan) [inline src/scripts/encodingsettings.js]: can you de-uglify at least this line?.. hard to tell what changed...

Comment (ploughpuff) [reply src/scripts/encodingsettings.js]: config.EncoderAppPath was changed to config.EncoderAppPathDisplay.

I'm no js expert so wouldn't know how to beautify this code..

Comment (ploughpuff) [reply src/scripts/encodingsettings.js]: Viewing the change without auto line wrap might help?

Comment (JustAMan) [reply src/scripts/encodingsettings.js]: de-uglification should look something like this:
```suggestion
        });
        page.querySelector("#chkHardwareEncoding").checked = config.EnableHardwareEncoding;
        $("#selectVideoDecoder", page).val(config.HardwareAccelerationType);
        $("#selectThreadCount", page).val(config.EncodingThreadCount);
        $("#txtDownMixAudioBoost", page).val(config.DownMixAudioBoost);
        page.querySelector(".txtEncoderPath").value = config.EncoderAppPathDisplay || "";
        $("#txtTranscodingTempPath", page).val(config.TranscodingTempPath || "");
        $("#txtVaapiDevice", page).val(config.VaapiDevice || "");
        page.querySelector("#selectH264Preset").value = config.H264Preset || "";
        page.querySelector("#txtH264Crf").value = config.H264Crf || "";
        page.querySelector("#chkEnableSubtitleExtraction").checked = config.EnableSubtitleExtraction || false;
        page.querySelector("#selectVideoDecoder").dispatchEvent(new CustomEvent("change", {
```

**Comment by ploughpuff** [comment]
Absolutely not.

File: src/scripts/encodingsettings.js [hunk 2]
```diff
@@ -90,8 +101,8 @@ define(["jQuery", "loading", "globalize", "dom"], function($, loading, globalize
         var page = this;
         ApiClient.getNamedConfiguration("encoding").then(function(config) {
             ApiClient.getSystemInfo().then(function(systemInfo) {
-                "External" == systemInfo.EncoderLocationType ? (page.querySelector(".fldEncoderPath").classList.add("hide"), page.querySelector(".txtEncoderPath").removeAttribute("required")) : (page.querySelector(".fldEncoderPath").classList.remove("hide"), page.querySelector(".txtEncoderPath").setAttribute("required", "required")), loadPage(page, config, systemInfo)
+                loadPage(page, config, systemInfo)
```
Comment (cvium) [inline src/scripts/encodingsettings.js]:
```suggestion
                loadPage(page, config, systemInfo);
```

File: src/scripts/encodingsettings.js [hunk 3]
```diff
@@ -4,7 +4,18 @@ define(["jQuery", "loading", "globalize", "dom"], function($, loading, globalize
...
+        $("#selectThreadCount", page).val(config.EncodingThreadCount);
+        $("#txtDownMixAudioBoost", page).val(config.DownMixAudioBoost);
+        page.querySelector(".txtEncoderPath").value = config.EncoderAppPathDisplay || "";
+        $("#txtTranscodingTempPath", page).val(config.TranscodingTempPath || "");
+        $("#txtVaapiDevice", page).val(config.VaapiDevice || "");
+        page.querySelector("#selectH264Preset").value = config.H264Preset || "";
+        page.querySelector("#txtH264Crf").value = config.H264Crf || "";
+        page.querySelector("#chkEnableSubtitleExtraction").checked = config.EnableSubtitleExtraction || false;
+        page.querySelector("#selectVideoDecoder").dispatchEvent(new CustomEvent("change", {
             bubbles: !0
```
Comment (cvium) [inline src/scripts/encodingsettings.js]:
```suggestion
            bubbles: true
```

File: src/scripts/encodingsettings.js [hunk 4]
```diff
@@ -4,8 +4,19 @@ define(["jQuery", "loading", "globalize", "dom"], function($, loading, globalize
...
+        $("#selectThreadCount", page).val(config.EncodingThreadCount);
+        $("#txtDownMixAudioBoost", page).val(config.DownMixAudioBoost);
+        page.querySelector(".txtEncoderPath").value = config.EncoderAppPathDisplay || "";
+        $("#txtTranscodingTempPath", page).val(config.TranscodingTempPath || "");
+        $("#txtVaapiDevice", page).val(config.VaapiDevice || "");
+        page.querySelector("#selectH264Preset").value = config.H264Preset || "";
+        page.querySelector("#txtH264Crf").value = config.H264Crf || "";
+        page.querySelector("#chkEnableSubtitleExtraction").checked = config.EnableSubtitleExtraction || false;
+        page.querySelector("#selectVideoDecoder").dispatchEvent(new CustomEvent("change", {
+             bubbles: true
```
Comment (cvium) [inline src/scripts/encodingsettings.js]: Hmm, where did that extra space come from....
```suggestion
            bubbles: true
```

**Comment by joshuaboniface** [comment]
This doesn't seem to show the value when the ffmpeg path is set via the CLI arg, e.g. on Debian. The box is still empty and red. Current master of main repo, and this branch on the web submodule.

**Pull request merged**